File: test/ruby/invalid_async.rb
#!/usr/bin/env ruby
require "bundler/setup"
Bundler.require

DocRaptor.configure do |dr|
  dr.username = "YOUR_API_KEY_HERE"
  dr.debugging = true
end

dr = DocRaptor::DocApi.new

response = dr.async_docs_post(test: true, document_content: "<html><body>Swagger Ruby</body></html>", name: "s" * 201, document_type: "pdf")

status_response = nil
30.times do
  status_response = dr.status_id_get(response.status_id)
  exit if status_response.status == "failed"
  sleep 1
end

abort "Did not receive failed validation within 30 seconds."

File: test/python/invalid_async.py
import docraptor
import time

docraptor.configuration.username = "YOUR_API_KEY_HERE"
docraptor.configuration.debug = True

default_api = docraptor.DefaultApi()

response = default_api.async_docs_post({"test": True, "document_content": "<html><body>Swagger Python</body></html>", "name": "s" * 201, "document_type": "pdf"})

status_response = None
for x in range(0, 30):
  status_response = default_api.status_id_get(response.status_id)
  if status_response.status == "failed":
    exit(0)
  time.sleep(1)

print "Exception expected, but not raised"
exit(1)
